modules/layers/src/geojson-layer/types.ts

```diff
@@ -0,0 +1,100 @@
+import {Accessor, Color, CompositeLayerProps, Unit} from '@deck.gl/core';
+import {FillProps, Polygon3DProps, StrokeProps} from '../types';
+
+/** All properties supported by GeoJsonLayer */
+export type GeoJsonLayerProps<DataT = any> = _GeoJsonLayerProps<DataT> & CompositeLayerProps<DataT>;
+
+/** Properties added by GeoJsonLayer */
+export type _GeoJsonLayerProps<DataT = any> = {
+  /**
+   * How to render Point and MultiPoint features in the data.
+   *
+   * Supported types are:
+   *  * `'circle'`
+   *  * `'icon'`
+   *  * `'text'`
+   *
+   * @default 'circle'
+   */
+  pointType?: string;
+
+  /** Type of line caps.
+   * If `true`, draw round caps. Otherwise draw square caps.
+   * @default false
+   */
+  lineCapRounded?: boolean;
+
+  /**
+   * If `true`, extrude the line in screen space (width always faces the camera).
+   * If `false`, the width always faces up.
+   *
+   * @default false
+   */
+  lineBillboard?: boolean;
+
+  /** @deprecated */
+  getRadius?: Accessor<DataT, number>;
+} & FillProps<DataT> &
+  StrokeProps<DataT> &
+  Polygon3DProps<DataT> &
```

Comment:
I would recommend you define all props explicitly here, just like how props are explicitly forwarded when GeoJsonLayer renders its sub-layers. Sharing prop types among layers will introduce the risk of declaring props that the layer does not actually support, and there is no way for the compiler to catch it.

Reply:
Ok, just wanted to avoid repetition.

OTOH, there is a way to ensure that Layer explicitly supports all properties defined in interface: properly typed defaultProps. Something like:
```ts
// some-layer.ts
type PropTypeDefsFor<PropT> = Record<keyof PropT, PropTypeDef>>;

const defaultProps = PropTypeDefsFor<_LayerProperties> = {
   // you're enforced to define _every_ property defined in _LayerProperties
   // of course layer implementation can ignore some props, but has to be explicit at least in 
   // one in layer source code
}
```
But that's probably story for other PR ...